datafusion/core/tests/user_defined/user_defined_scalar_functions.rs

```diff
@@ -498,6 +503,83 @@ async fn test_user_defined_functions_zero_argument() -> Result<()> {
     Ok(())
 }
 
+#[derive(Debug)]
+struct CastToI64UDF {
+    signature: Signature,
+}
+
+impl CastToI64UDF {
+    fn new() -> Self {
+        Self {
+            signature: Signature::any(1, Volatility::Immutable),
+        }
+    }
+}
+
+impl ScalarUDFImpl for CastToI64UDF {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+    fn name(&self) -> &str {
+        "cast_to_i64"
+    }
+    fn signature(&self) -> &Signature {
+        &self.signature
+    }
+    fn return_type(&self, _args: &[DataType]) -> Result<DataType> {
+        Ok(DataType::Int64)
+    }
+    // Wrap with Expr::Cast() to Int64
+    fn simplify(&self, args: &[Expr]) -> Result<Simplified> {
+        let dfs = DFSchema::new_with_metadata(
```

Comment:
This is a great example, thank you @jayzhan211 

It seems to me that the UDF can't always know what the input schema would be, and thus the schema should be passed in.

Something like this

```rust
    fn simplify(&self, args: &[Expr], schema: &DFSchema) -> Result<Simplified> {
...
     }
```

What do you think?

Reply:
SimpliyContext has DFSchemaRef, so I change the type to DFSchemaRef, I think it is also fine